cmd/statreceiver/common.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"log"
+	"time"
+)
+
+// Deliver kicks off a goroutine that reads packets from s and delivers them
+// to p.
+func Deliver(s Source, p PacketDest) {
+	go func() {
+		for {
```

Comment:
There is no way out of this infinite loop. Is this intended? Any chance for goroutine leak?

Reply:
this is intended, yeah. when the lua script calls deliver it's supposed to start a thread for the remainder of the lifetime of the process that hooks the source to the destination